### PR TITLE
Deprecate EqualType

### PR DESCRIPTION
**Pull request by VincentLanglet**
## Subject

Related to https://github.com/sonata-project/form-extensions/issues/67
But not close completely.

## Changelog

```markdown
### Deprecated
- Deprecate EqualType in favor of SonataAdmin EqualOperatorType
```

### CODE REVIEW

**Comment by greg0ire** [comment]
Tests relying on the old type should now use the new type, or be marked as deliberately doing so by using `@group legacy`

**Comment by VincentLanglet** [comment]
@greg0ire Done !

**Comment by VincentLanglet** [comment]
@greg0ire I rebased the 1.x to fix phpstan.

Wanna take a look @core23 @phansys ?

**Comment by greg0ire** [comment]
I often forget to ask it, but our policy says we are supposed to require an upgrade note when a new deprecation is introduced.

**Comment by VincentLanglet** [comment]
@greg0ire You mean updating the UPGRADE-2.x.md file ?

**Comment by greg0ire** [comment]
I think it should be 1.x in our case, here is how it usually looks like: https://github.com/sonata-project/SonataAdminBundle/blob/3.x/UPGRADE-3.x.md I see that it does not exist, so please create it.

**Comment by VincentLanglet** [comment]
@greg0ire Done

File: UPGRADE-1.x.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+UPGRADE 1.x
+===========
+
+## Deprecated EqualType form
+
+If you use `Sonata\Form\Type\EqualType`, you should use `Sonata\AdminBundle\Form\Type\Operator` instead.
```
Comment (core23) [inline UPGRADE-1.x.md]:
```suggestion
If you use `Sonata\Form\Type\EqualType`, you should use `Sonata\AdminBundle\Form\Type\Operator\EqualOperatorType` instead.
```

**Comment by VincentLanglet** [comment]
Nice catch @core23, I'm making so much little mistake...

**Comment by greg0ire** [comment]
Thanks @VincentLanglet !

**Pull request merged**